cfgov/ask_cfpb/models/pages.py

```diff
@@ -241,6 +244,211 @@ def page_js(self):
+        if not search_term or len(unquote(search_term)) == 1:
+            count = self.query_base.count()
+            sqs = self.query_base
+            search_message = self.results_message(count, heading, search_term)
+        else:
+            sqs = self.query_base.filter(content=search_term)
+            count = sqs.count()
+            if count == 0:
+                # No results, so let's try to suggest a better query
+                suggestion = SearchQuerySet().models(
+                    AnswerPage).spelling_suggestion(search_term)
+                if suggestion == search_term:
+                    suggestion = None
+                elif (request.GET.get('correct', '1') == '1' and
+                        flag_enabled('ASK_SEARCH_TYPOS', request=request)):
+                    sqs = self.query_base.filter(content=suggestion)
+                    search_term, suggestion = suggestion, search_term
+                    count = sqs.count()
+            context['pages'] = sqs
+            search_message = self.results_message(count, heading, search_term)
+        paginator = Paginator(sqs, 10)
+        page_number = validate_page_number(request, paginator)
+        pages = paginator.page(page_number)
+        context.update({
+            'count': count,
+            'search_term': search_term,
+            'results_message': search_message,
+            'pages': pages,
+            'paginator': paginator,
+            'current_page': page_number,
+            'get_secondary_nav_items': self.get_nav_items,
+        })
+        return TemplateResponse(
+            request,
+            self.get_template(request),
+            context)
+
+    @route(r'^(?P<category>[^/]+)/$')
+    def portal_category_page(self, request, **kwargs):
+        context = self.get_context(request)
+        category_slug = kwargs.get('category')
+        self.portal_category = self.category_map.get(category_slug)
+        heading = self.get_heading(self.portal_category)
+        sqs = self.query_base.filter(
+            portal_categories=self.portal_category.heading)
+        search_term = request.GET.get('search_term', '').strip()
+        if not search_term or len(unquote(search_term)) == 1:
+            count = sqs.count()
+            search_message = self.results_message(count, heading, search_term)
+        else:
+            sqs = sqs.filter(content=search_term)
+            count = sqs.count()
+            if count == 0:
+                # No results, so let's try to suggest a better query
+                suggestion = SearchQuerySet().models(
+                    AnswerPage).spelling_suggestion(search_term)
+                if suggestion == search_term:
+                    suggestion = None
+                elif (request.GET.get('correct', '1') == '1' and
+                        flag_enabled('ASK_SEARCH_TYPOS', request=request)):
+                    sqs = self.query_base.filter(content=suggestion)
```

Comment:
This is no longer filtering with the provided category.  I think refactoring the search code as I did in https://github.com/cfpb/cfgov-refresh/tree/refactor-search-suggestion (not functional yet, I know, but as a proof of concept) might make this code more workable and easier to test.

Reply:
I fixed the filter, and we can refactor.